src/utilities/modules/dispensary/pay.js

```diff
@@ -73,27 +78,30 @@ export const pay = (
     receiptDescription
   );
 
-  createRecord(UIDatabase, 'ReceiptLine', receipt, script, cashAmount);
+  // Create a receipt line for the full non-credit amount, if any.
+  if (cashAmount) createRecord(UIDatabase, 'ReceiptLine', receipt, script, cashAmount);
 
-  // When using credit, create a CustomerCreditLine and ReceiptLine for each
-  // source of credit being used.
   let creditToAllocate = creditAmount;
+
   if (usingCredit) {
-    // Iterate over all the patients CustomerCredits, taking credit
-    // from each with available credit, until all has been allocated.
-    patient.customerCredits.some(customerCredit => {
-      const creditFromCustomerCredit = customerCredit.outstanding;
+    // Create a receipt line for the full credit amount, if any.
+    createRecord(UIDatabase, 'ReceiptLine', receipt, script, creditAmount);
 
-      if (creditFromCustomerCredit >= 0) return false;
+    // Iterate over all the patients credit sources, taking credit from each with
+    // available credit, by reducing the outstanding value until all has been allocated.
+    patient.creditSources.some(creditSource => {
```

Comment:
Hmm, what's going on here with `some`? I'm guessing to shortcut once `creditToAllocate` is `0`? I can see why you'd want to shortcut, but I feel like performing side-effects in `some` might defeat the purpose of using a functional construct here? I think a loop might actually be the better way of doing this for readability + performance?

Reply:
`some` v `while` with indicies v `for each with a break` -- they're all not right so I'm not touching it and walking away ahhhhhh